src/test/java/htsjdk/samtools/cram/build/ContainerFactoryTest.java

```diff
@@ -1,148 +1,297 @@
-
-        // build a container with the max records per slice
-
-        final List<CramCompressionRecord> records = CRAMStructureTestUtil.getSingleRefRecords(recordsPerSlice, 0);
-        final long dummyByteOffset = 0;
-        final Container container = factory.buildContainer(records, dummyByteOffset);
+    @DataProvider(name="singleContainerSliceDistribution")
+    private Object[][] getSingleContainerSliceDistribution() {
+        final int RECORDS_PER_SLICE = 100;
+        return new Object[][] {
+                // samRecords, records/slice, slices/container, expected refContext id, expected slice count,
+                // expected slice refContexts, expected record count/slice
+                {
+                        // 1 full single-ref slice with 1 rec
+                        CRAMStructureTestHelper.createSAMRecordsMapped(1, 0),
+                        RECORDS_PER_SLICE, 1,
+                        new ReferenceContext(0),
+                        1, Arrays.asList(1), Arrays.asList(new ReferenceContext(0))
+                },
+                {
+                        // 1 full single-ref slice with 1 rec, but allow > 1 slices/container
+                        CRAMStructureTestHelper.createSAMRecordsMapped(1, 0),
+                        RECORDS_PER_SLICE, 2,
+                        new ReferenceContext(0),
+                        1, Arrays.asList(1), Arrays.asList(new ReferenceContext(0))
+                },
+                {
+                        // 1 full single-ref slice with RECORDS_PER_SLICE - 1 records
+                        CRAMStructureTestHelper.createSAMRecordsMapped(RECORDS_PER_SLICE - 1, 0),
+                        RECORDS_PER_SLICE, 1,
+                        new ReferenceContext(0),
+                        1, Arrays.asList(RECORDS_PER_SLICE - 1), Arrays.asList(new ReferenceContext(0))
+                },
+                {
+                        // 1 full single-ref slice with RECORDS_PER_SLICE records
+                        CRAMStructureTestHelper.createSAMRecordsMapped(RECORDS_PER_SLICE, 0),
+                        RECORDS_PER_SLICE, 1,
+                        new ReferenceContext(0),
+                        1, Arrays.asList(RECORDS_PER_SLICE), Arrays.asList(new ReferenceContext(0))
+                },
+                {
+                        // 2 single-ref slices, one with RECORDS_PER_SLICE records, one with 1 record
+                        CRAMStructureTestHelper.createSAMRecordsMapped(RECORDS_PER_SLICE + 1, 0),
+                        RECORDS_PER_SLICE, 2,
+                        new ReferenceContext(0),
+                        2, Arrays.asList(RECORDS_PER_SLICE, 1),
+                        Arrays.asList(new ReferenceContext(0), new ReferenceContext(0))
+                },
+                {
+                        // 2 full single-ref slices, each with RECORDS_PER_SLICE records
+                        CRAMStructureTestHelper.createSAMRecordsMapped(RECORDS_PER_SLICE * 2, 0),
```

Comment:
I would also check `RECORDS_PER_SLICE * 2 - 1` and `RECORDS_PER_SLICE * 2 + 1` as boundary conditions, as the single-slice cases alone are not sufficiently reassuring.

Reply:
Done.